docs/source/how-to/configure-workflows/onnx-graph-surgeon.md

```diff
@@ -0,0 +1,542 @@
+    input: ["renamed_input1", "renamed_input2"]
+    output: ["add_output"]
+  }
+}
+```
+
+### `RenameOutputs`
+
+#### Description
+
+Renames specific outputs in the ONNX model.
+
+#### Configurations
+
+- `old_names`: List of original output names to rename.
+- `new_names`: List of new output names.
+
+#### Example
+
+Initial ONNX model graph:
+
+```
+graph {
+  node {
+    op_type: "Add"
+    input: ["input1", "input2"]
+    output: ["output1"]
+  }
+}
+```
+
+After applying:
+
+```json
+{
+    "type": "GraphSurgeries",
+    "surgeries": [
+        {
+            "surgeon": "RenameOutputs",
+            "old_names": ["output1"],
+            "new_names": ["renamed_output1"]
+        }
+    ]
+}
+```
+
+Transformed ONNX model graph:
+
+```
+graph {
+  node {
+    op_type: "Add"
+    input: ["input1", "input2"]
+    output: ["renamed_output1"]
+  }
+}
+```
+
+### `InferShapes`
+
+#### Description
+
+Performs shape inference on the ONNX model to populate missing shape information.
+
+#### Example
+
+Initial ONNX model graph:
+
+```
+graph {
+  input: "input1" (FLOAT) shape: [1]
+  input: "input2" (FLOAT) shape: [1]
+  node {
+    op_type: "Add"
+    input: ["input1", "input2"]
+    output: ["add_output"]
+  }
+  output: "add_output" (FLOAT)
+}
+```
+
+After applying:
+
+```json
+{
+    "type": "GraphSurgeries",
+    "surgeries": [
+        {
+            "surgeon": "InferShapes"
+        }
+    ]
+}
+```
+
+Transformed ONNX model graph (with inferred shapes):
+```
+graph {
+  input: "input1" (FLOAT) shape: [1]
+  input: "input2" (FLOAT) shape: [1]
+  node {
+    op_type: "Add"
+    input: ["input1", "input2"]
+    output: ["add_output"]
+  }
+  output: "add_output" (FLOAT) shape: [1]
+}
+```
+
+
+### `RemoveShapes`
+
+#### Description
+
+Removes all shape information (`value_info`) from the ONNX model.
+
+#### Example
+
+Initial ONNX model graph:
+
+```
+graph test-model {
+  # Inputs
+  input: "input1" (FLOAT) shape: [1]
+  input: "input2" (FLOAT) shape: [1]
+
+  # Value Info
+  value_info: "intermediate" (FLOAT) shape: [1]
+
+  # Nodes
+  node {
+    op_type: "Add"
+    name: "Add"
+    input: ["input1", "input2"]
+    output: ["intermediate"]
+  }
+
+  node {
+    op_type: "Relu"
+    name: "Relu"
+    input: ["intermediate"]
+    output: ["output1"]
+  }
+
+  # Outputs
+  output: "output1" (FLOAT) shape: [1]
+}
+```
+
+After applying:
+
+```json
+{
+    "type": "GraphSurgeries",
+    "surgeries": [
+        {
+            "surgeon": "RemoveShapes"
+        }
+    ]
+}
+```
+
+Transformed ONNX model graph:
+
+```
+graph test-model {
+  # Inputs
+  input: "input1" (FLOAT) shape: [1]
+  input: "input2" (FLOAT) shape: [1]
+
+  # Nodes
+  node {
+    op_type: "Add"
+    name: "Add"
+    input: ["input1", "input2"]
+    output: ["intermediate"]
+  }
+
+  node {
+    op_type: "Relu"
+    name: "Relu"
+    input: ["intermediate"]
+    output: ["output1"]
+  }
+
+  # Outputs
+  output: "output1" (FLOAT) shape: [1]
+}
+```
+
+
+### `ReorderInputs`
```

Comment:
I have a question about this. do you know if this is used to permute the model inputs or to permute node inputs? 

If we do it for both, then it behaves like we are changing the names of the inputs. Since onnx operators have a fixed input order, I wonder how this is an input "reorder".

Reply:
Yeah good call. Renaming input won't change the input shape while reorder should do. Let me double check the use case of it.

Reply:
I removed node reorder for now. Waiting for more details.

Reply:
Sounds good. Thanks!